### ISSUE TITLE
EKS cluster unreachable

### ISSUE THREAD

**Issue by burner1024** [containers/podman-desktop#10479]
### Bug description

Can't access EKS cluster from `~/.kube/config`.
The context is detected correctly. However, the ui displays "UNREACHABLE". 
Hover shows `Error: spawn aws ENOENT`.
`aws` is in PATH, and I can run it just fine. Not sure where PD searches for it, and why doesn't it use correct `PATH`.
In fact, not sure why PD even needs it in the first place to connect to EKS. `kubectl` (which is also in `PATH` and works fine) should be enough, I think.

### Operating system

ubuntu 24.04

### Installation Method

Flathub (Linux)

### Version

1.15.0

### Steps to reproduce

Just install and launch.

### Relevant log output

```shell

```

**Comment by benoitf** [comment]
hello, I suspect that as Podman Desktop is running in flathub in your case, it has no access to the binary installed on your system as Podman Desktop is sandboxed

You might give a try to the .tgz (unpack the archive and launch the binary from there) version of Podman Desktop

https://github.com/podman-desktop/podman-desktop/releases/download/v1.15.0/podman-desktop-1.15.0.tar.gz

**Comment by burner1024** [comment]
Yes, it's flatpak. But why does it call `aws`? And what's the proper solution for this?

**Comment by benoitf** [comment]
Podman Desktop is checking the state of clusters listed in the kubeconfig file

as the file is referencing `aws` binary it'll try to execute it.

workaround would be to use the binary (non flatpak)

proper solution would be that the kubernetes client being used could delegate the calls to the host (your computer vs the sandboxed environment) when detecting it's running within flatpak

**Comment by burner1024** [comment]
OK, this seems like a strange state of affairs.
But thank you for clarification.

**Comment by odockal** [comment]
@benoitf Is there something more we can do with the issue, or the was the explanation sufficient?

**Comment by benoitf** [comment]
in Podman Desktop directly we can't do much if the library is not letting intercept or handling things, as it's the kubernetes client library that is calling the binary

Explore things like: can we intercept the calls to a binary from this library ? if yes, adds the `flatpak-spawn` as prefix to execute the command on the host

**Comment by GuusDeGraeve** [comment]
I'm having this same issue on MacOS, which is not using flathub.

**Comment by benoitf** [comment]
@GuusDeGraeve in your case, I suspect that the binary is in your 'shell' path but the macOS applications are launched without a login shell, meaning that the default PATH is like /sbin, /bin, /usr/bin etc and for example if you're using brew, aws binary will not be seen as it's in a folder not in the default PATH

**Comment by GuusDeGraeve** [comment]
> [@GuusDeGraeve](https://github.com/GuusDeGraeve) in your case, I suspect that the binary is in your 'shell' path but the macOS applications are launched without a login shell, meaning that the default PATH is like /sbin, /bin, /usr/bin etc and for example if you're using brew, aws binary will not be seen as it's in a folder not in the default PATH

Thank you for the instant reply, I tried this and now it is working 🎉:
```
❯ export PATH=$PATH:/opt/homebrew/bin/
❯ open /Applications/Podman\ Desktop.app
```

**Comment by serhiiKalchenko** [comment]
Guys, why do you still have the main way to download your product through flatpak on your official website, if it doesn't work through it (Linux, flatpak, AWS EKS)? I got a negative experience with your product, lost a lot of time to find out that this is the reason (you need to install it from the archive). That's the new version 1.16.0.

[https://podman-desktop.io/downloads/linux](https://podman-desktop.io/downloads/linux)
The button 'Download Now' downloads **podman-desktop-*.flatpak**

**Comment by feloy** [comment]
> > [@GuusDeGraeve](https://github.com/GuusDeGraeve) in your case, I suspect that the binary is in your 'shell' path but the macOS applications are launched without a login shell, meaning that the default PATH is like /sbin, /bin, /usr/bin etc and for example if you're using brew, aws binary will not be seen as it's in a folder not in the default PATH
> 
> Thank you for the instant reply, I tried this and now it is working 🎉:
> 
> ```
> ❯ export PATH=$PATH:/opt/homebrew/bin/
> ❯ open /Applications/Podman\ Desktop.app
> ```

If you don't want to have to extend your PATH variable, you can also modify your ~/.kube/config file to specify the complete path of `eks`

**Comment by josecastronieto** [comment]
I used to run Podman Desktop  from the binary to circumvent the aws access problem. However I went back to flatpak because I wanted to get the updates (plus it is supposed to be the recommended way).
The workaround I found requires installing the AWS CLI inside the Flatpak sandbok. However this has to be done in a writeable directory within the sandbox and the path must be modified. 
Here are the steps:

1. In the terminal, run: `flatpak run --devel --command=sh io.podman_desktop.PodmanDesktop`
2. Inside the shell, install the AWS CLI in writeable directory (`/var/data/`):
```
cd /tmp
curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip"
unzip awscliv2.zip
./aws/install -i /var/data/aws-cli -b /var/data/aws-cli/bin
export PATH=/var/data/aws-cli/bin:$PATH
```
3. Test also within the shell:
```
which aws
aws --version
```
4. Exit the Flatpak sandbox shell and set the PATH persistently for the Flatpak app. 
Run `flatpak override --user --env=PATH=/var/data/aws-cli/bin:/app/bin:/usr/bin io.podman_desktop.PodmanDesktop`

Now it should work.
